client/state/data-layer/wpcom/posts/revisions/index.js

```diff
@@ -18,56 +18,57 @@ import { registerHandlers } from 'state/data-layer/handler-registry';
  * Dispatches returned error from post revisions request
  *
  * @param {Function} dispatch Redux dispatcher
- * @param {Object} action Redux action
  * @param {String} action.siteId of the revisions
  * @param {String} action.postId of the revisions
  * @param {Object} rawError from HTTP request
- * @returns {Object} the dispatched action
+ *
+ * @return {Object} the dispatched action
  */
-export const receiveError = ( { dispatch }, { siteId, postId }, rawError ) =>
-	dispatch( receivePostRevisionsFailure( siteId, postId, rawError ) );
+export const receiveError = ( { siteId, postId }, rawError ) =>
+	receivePostRevisionsFailure( siteId, postId, rawError );
 
 /**
  * Dispatches returned post revisions
  *
- * @param {Function} dispatch Redux dispatcher
  * @param {Object} action Redux action
  * @param {String} action.siteId of the revisions
  * @param {String} action.postId of the revisions
  * @param {Object} response from the server
  * @param {Object} response.diffs raw data containing a set of diffs for the site & post
+ *
+ * @return {Array} An array of Redux actions
  */
-export const receiveSuccess = ( { dispatch }, { siteId, postId }, response ) => {
-	dispatch( receivePostRevisionsSuccess( siteId, postId ) );
-	dispatch( receivePostRevisions( { siteId, postId, ...response } ) );
+export const receiveSuccess = ( { siteId, postId }, response ) => {
+	return [
+		receivePostRevisionsSuccess( siteId, postId ),
```

Comment:
The `POST_REVISIONS_REQUEST_{SUCCESS,FAILURE}` actions are used only to update the `state.posts.revisions.requesting` reducer, whose values are not used anywhere.

That means that we can remove the reducer and both actions, leaving only `POST_REVISIONS_RECEIVE` handling the responses. We can simplify this handler, too, to dispatch only one action, and set the `onError` handler to `noop`.

Reply:
I pushed the changes I recommended (`SUCCESS` and `FAILURE` actions and `requesting` reducer removed) in 3d45459. Going to merge after verifying the change tests well.